sqlglot/parser.py

```diff
@@ -2380,6 +2420,14 @@ def _parse_row(self) -> t.Optional[exp.RowFormatSerdeProperty | exp.RowFormatDel
             return None
         return self._parse_row_format()
 
+    def _parse_serde_properties(self, with_=None) -> t.Optional[exp.SerdeProperties]:
+        with_ = self._match_text_seq("WITH") or with_
+        if not self._match(TokenType.SERDE_PROPERTIES):
+            return None
```

Comment:
What if you consume `WITH` and then you don't find `SERDE_PROPERTIES`? Don't we need to retreat?